### PR TITLE
Update templates to support the new pattern used for integration tests

### PR DESCRIPTION
**Pull request by javiercn**
I tried running the EF tools with both patterns to verify that they both work. Below is the output

```
C:\work\template-tests\pattern> cat .\Program.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Logging;

namespace pattern
{
    public class Program
    {
        public static void Main(string[] args)
        {
            CreateWebHostBuilder(args).Build().Run();
        }

        public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>();
    }
}
```

```
C:\work\template-tests\pattern> dotnet ef migrations list
info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[0]
      User profile is available. Using 'C:\Users\jacalvar\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
info: Microsoft.EntityFrameworkCore.Infrastructure[10403]
      Entity Framework Core 2.1.0-preview1-27373 initialized 'ApplicationDbContext' using provider 'Microsoft.EntityFrameworkCore.Sqlite' with options: None
00000000000000_CreateIdentitySchema
```

```
C:\work\template-tests\pattern> code . .\Program.cs
```

````
C:\work\template-tests\pattern> cat .\Program.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Logging;

namespace pattern
{
    public class Program
    {
        public static void Main(string[] args)
        {
            BuildWebHost(args).Run();
        }

        public static IWebHost BuildWebHost(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .Build();
    }
}

C:\work\template-tests\pattern> dotnet ef migrations list
info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[0]
      User profile is available. Using 'C:\Users\jacalvar\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
info: Microsoft.EntityFrameworkCore.Infrastructure[10403]
      Entity Framework Core 2.1.0-preview1-27373 initialized 'ApplicationDbContext' using provider 'Microsoft.EntityFrameworkCore.Sqlite' with options: None
00000000000000_CreateIdentitySchema
```

### CODE REVIEW

File: src/Microsoft.DotNet.Web.ProjectTemplates.2.1/content/EmptyWeb-CSharp/Program.cs [hunk 1]
```diff
@@ -14,12 +14,11 @@ public class Program
     {
         public static void Main(string[] args)
         {
-            BuildWebHost(args).Run();
+            CreateWebHostBuilder(args).Build().Run();
         }
 
-        public static IWebHost BuildWebHost(string[] args) =>
```
Comment (ajcvickers) [inline src/Microsoft.DotNet.Web.ProjectTemplates.2.1/content/EmptyWeb-CSharp/Program.cs]: Won't this break EF tooling? /cc @bricelam

Comment (Eilon) [reply src/Microsoft.DotNet.Web.ProjectTemplates.2.1/content/EmptyWeb-CSharp/Program.cs]: I believe EF tooling was updated by @javiercn .

Comment (Eilon) [reply src/Microsoft.DotNet.Web.ProjectTemplates.2.1/content/EmptyWeb-CSharp/Program.cs]: https://github.com/aspnet/EntityFrameworkCore/pull/10100

**Comment by SteveSandersonMS** [comment]
@Eilon Any changes to the .NET parts of the project templates should be applied to the SPA ones at the same time, yes.

File: src/Microsoft.DotNet.Web.ProjectTemplates.2.1/content/EmptyWeb-CSharp/Program.cs (hunk 1 above)
Comment (ajcvickers) [reply src/Microsoft.DotNet.Web.ProjectTemplates.2.1/content/EmptyWeb-CSharp/Program.cs]: Okay, so EF tooling now supports both patterns? So, this change will work with EF Core 2.1, but not with 2.0, but an app created using the 2.0 pattern will continue to work with 2.1? If so, I think that's okay.

Comment (javiercn) [reply src/Microsoft.DotNet.Web.ProjectTemplates.2.1/content/EmptyWeb-CSharp/Program.cs]: Yep

**Comment by javiercn** [comment]
@SteveSandersonMS I've updated the spa templates with the new pattern.
@Eilon I've also updated the F# templates.

File: src/Microsoft.DotNet.Web.ProjectTemplates.2.1/content/EmptyWeb-FSharp/Program.fs [hunk 2]
```diff
@@ -13,14 +13,13 @@ open Microsoft.Extensions.Logging
 module Program =
     let exitCode = 0
 
-    let BuildWebHost args =
+    let CreateWebHostBuilder args =
```
Comment (Eilon) [inline src/Microsoft.DotNet.Web.ProjectTemplates.2.1/content/EmptyWeb-FSharp/Program.fs]: Very... nice?? 😄

**Pull request merged**